src/main/java/com/google/changestreams/sample/bigquery/changelog/fullschema/SpannerChangeStreamsToBigQuery.java

```diff
@@ -0,0 +1,326 @@
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.changestreams.sample.bigquery.changelog.fullschema;
+
+import com.google.api.services.bigquery.model.TableRow;
+import com.google.changestreams.sample.bigquery.changelog.fullschema.model.Mod;
+import com.google.changestreams.sample.bigquery.changelog.fullschema.schemautils.BigQueryUtils;
+import com.google.cloud.Timestamp;
+import com.google.cloud.teleport.v2.cdc.dlq.DeadLetterQueueManager;
+import com.google.cloud.teleport.v2.cdc.dlq.StringDeadLetterQueueSanitizer;
+import com.google.cloud.teleport.v2.coders.FailsafeElementCoder;
+import com.google.cloud.teleport.v2.transforms.DLQWriteTransform;
+import com.google.cloud.teleport.v2.values.FailsafeElement;
+import org.apache.beam.runners.dataflow.options.DataflowPipelineOptions;
+import org.apache.beam.sdk.Pipeline;
+import org.apache.beam.sdk.PipelineResult;
+import org.apache.beam.sdk.coders.StringUtf8Coder;
+import org.apache.beam.sdk.io.gcp.bigquery.BigQueryIO;
+import org.apache.beam.sdk.io.gcp.bigquery.BigQueryIO.Write;
+import org.apache.beam.sdk.io.gcp.bigquery.BigQueryIO.Write.CreateDisposition;
+import org.apache.beam.sdk.io.gcp.bigquery.InsertRetryPolicy;
+import org.apache.beam.sdk.io.gcp.bigquery.WriteResult;
+import org.apache.beam.sdk.io.gcp.spanner.SpannerConfig;
+import org.apache.beam.sdk.io.gcp.spanner.SpannerIO;
+import org.apache.beam.sdk.io.gcp.spanner.changestreams.model.DataChangeRecord;
+import org.apache.beam.sdk.options.PipelineOptionsFactory;
+import org.apache.beam.sdk.options.ValueProvider;
+import org.apache.beam.sdk.transforms.*;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.beam.sdk.values.PCollectionList;
+import org.apache.beam.sdk.values.PCollectionTuple;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.Set;
+
+/**
+ * This pipeline ingests {@link DataChangeRecord} from Spanner Change Stream. The
+ * {@link DataChangeRecord} is then broken into {@link Mod}, which converted into {@link TableRow}
+ * and inserted into BigQuery table.
+ */
+public class SpannerChangeStreamsToBigQuery {
+
+  /**
+   * String/String Coder for {@link FailsafeElement}.
+   */
+  public static final FailsafeElementCoder<String, String> FAILSAFE_ELEMENT_CODER =
+    FailsafeElementCoder.of(StringUtf8Coder.of(), StringUtf8Coder.of());
+
+  private static final Logger LOG = LoggerFactory.getLogger(SpannerChangeStreamsToBigQuery.class);
+
+  /**
+   * Main entry point for executing the pipeline.
+   *
+   * @param args The command-line arguments to the pipeline.
+   */
+  public static void main(String[] args) {
```

Comment:
This class has a `main` and we also a `Main` class. This is confusing. Could you rename the original `Main` class to something more meaningful now that we have two classes that expose main in this package?

Reply:
Renamed SpannerChangeStreamsToBigQuerySerialized, let me know if you have a better name :)